### PR TITLE
HPCC-19267 Add a flushDiskCache option option switchable by //version in to regression suite

### PR DESCRIPTION
**Pull request by AttilaVamos**
Implement flushDiskCache in versioning.
Add --flushDiskCache as a command line parameter
Handle flushDiskcache on ecl execution level
Add user privilege check
Extend error handling and error messages

Update README.rst

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested manually on a local VM with and without sudo privileges.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@richardkchapman please review

File: testing/regress/hpcc/common/logger.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def emit(self, record):
                 msg = self.format(record)
                 stream = self.stream
                 isBuffer = hasattr(record, 'filebuffer')
-                toSort = hasattr(record,  'filesort')
+                #toSort = hasattr(record,  'filesort')
```
Comment (richardkchapman) [inline testing/regress/hpcc/common/logger.py]: Why is this commented out?

File: testing/regress/hpcc/util/ecl/file.py [hunk 2]
```diff
@@ -498,6 +502,13 @@ def setElapsTime(self,  time):
         self.elapsTime = time
 
     def setJobnameVersion(self,  version):
+        # Overrides the global flushDiskCache parameter if --pq <= 1
+        # There is no sense to clear disk cache if same test runnnig parallel by versioning
```
Comment (richardkchapman) [inline testing/regress/hpcc/util/ecl/file.py]: typo - runnnig

**Comment by richardkchapman** [comment]
@AttilaVamos A couple of minor questions

File: testing/regress/hpcc/common/logger.py (hunk 1 above)
Comment (AttilaVamos) [reply testing/regress/hpcc/common/logger.py]: That was an unused variable, removed. 
(Long time ago there was a forced sort for log lines when parallel queries used. It is already improved, the sort replaced by speed effective data structure, but somehow this line left there. Now the Python IDE put a notification to that line, I commented out before test and forget to remove before push.)

File: testing/regress/hpcc/util/ecl/file.py (hunk 2 above)
Comment (AttilaVamos) [reply testing/regress/hpcc/util/ecl/file.py]: Changed

**Comment by AttilaVamos** [comment]
@richardkchapman updated, please check and merge.

**Pull request merged**